pynxtools/dataconverter/readers/stm/stm_file_parser.py

```diff
@@ -0,0 +1,383 @@
+#
+
+
+import os
+from typing import Any, Dict
+import logging
+import re
+import numpy as np
+import nanonispy as nap
+
+from pynxtools.dataconverter.readers.stm.stm_helper import (nested_path_to_slash_separated_path,
+                                                            transform, fill_template_from_eln_data,
+                                                            work_out_overwriteable_field,
+                                                            link_implementation, UNIT_TO_SKIP)
+
+
+logging.basicConfig(level=logging.INFO, format='%(levelname)s - %(message)s')
+
+
+def has_separator_char(key, sep_char_li):
+    """
+    Check string or key whether the separator char provided in
+    'Separator Char List' exist or not.
+    """
+    bool_k = [x in sep_char_li for x in key]
+    return np.any(bool_k)
+
+
+# pylint: disable=invalid-name
+class STM_Nanonis():
+    """Specific class for stm reader from nanonis company.
+
+    """
+
+    def __init__(self, file_name):
+        """Construct
+        """
+
+        self.file_name = file_name
+
+    def get_nested_dict_from_concatenated_key(self, data_dict, dict_to_map_path=None,
+                                              sep_chars=None):
+        """
+        Create nested dict. If key are concateneted with '_', '>' split the key and
+        construct nested dict. For example, {'x1': {'x2': {'x3': {'x4': {'x5': 3}}}}
+        from 'x1_x2_x3_x4>x5:3'
+        """
+        if dict_to_map_path is not None:
+            spreaded_dict = dict_to_map_path
+        else:
+            spreaded_dict: Dict[str, Any] = {}
+        if sep_chars is None:
+            sep_chars = ['_', '>']
+        for d_key, d_val in data_dict.items():
+            if has_separator_char(d_key, sep_chars):
+                # Find out which separator char exist there
+                for k_c in d_key:
+                    if k_c in sep_chars:
+                        sep_char = k_c
+                        break
+                l_key, r_key = d_key.split(sep_char, 1)
+                if not has_separator_char(r_key, sep_chars):
+                    if l_key not in spreaded_dict:
+                        spreaded_dict[l_key]: Dict[str, Any] = {}
+                    spreaded_dict[l_key][r_key] = d_val
+                else:
+                    if l_key in spreaded_dict:
+                        spreaded_dict[l_key] = self.get_nested_dict_from_concatenated_key(
+                            {r_key: d_val}, dict_to_map_path=spreaded_dict[l_key])
+                    else:
+                        spreaded_dict[l_key]: Dict[str, Any] = {}
+                        spreaded_dict[l_key] = self.get_nested_dict_from_concatenated_key(
+                            {r_key: d_val}, dict_to_map_path=spreaded_dict[l_key])
+            else:
+                spreaded_dict[d_key] = d_val
+
+        return spreaded_dict
+
+    def convert_key_to_unit_and_entity(self, key, val, start_bracket='', end_bracket=''):
```

Comment:
I think one could still cleaner separate the different (currently two (sub-)parsers in this example, for a first round I think what we have lgtm though @RubelMozumder